packages/teleterm/src/ui/services/connectionTracker/trackedConnectionOperationsFactory.ts

```diff
@@ -0,0 +1,121 @@
+import { routing } from 'teleterm/ui/uri';
+import { ClustersService } from 'teleterm/ui/services/clusters';
+import { WorkspacesService } from 'teleterm/ui/services/workspacesService';
+import {
+  getGatewayDocumentByConnection,
+  getServerDocumentByConnection,
+} from './trackedConnectionUtils';
+import {
+  TrackedConnection,
+  TrackedGatewayConnection,
+  TrackedServerConnection,
+} from './types';
+
+export class TrackedConnectionOperationsFactory {
+  constructor(
+    private _clustersService: ClustersService,
+    private _workspacesService: WorkspacesService
+  ) {}
+
+  create(connection: TrackedConnection): TrackedConnectionOperations {
+    switch (connection.kind) {
+      case 'connection.server':
+        return this.getConnectionServerOperations(connection);
+      case 'connection.gateway':
+        return this.getConnectionGatewayOperations(connection);
+    }
+  }
```

Comment:
This will have to be extended for any kind of connection that we want to support and for now we support SSH servers and DB proxy, do I understand this right?